### PR TITLE
Added checks for 'no file chosen' and 'maxFileSize exceeded'

### PR DESCRIPTION
**Pull request by savsch**
Added checks for 'no file chosen' and, more importantly, 'maxFileSize exceeded' on the page served.

### CODE REVIEW

File: http-server-upload.js [hunk 1]
```diff
@@ -208,6 +208,7 @@ server.on('request', (req, res) => {
 
     });
   } else {
+    let formId="fileinput";
```
Comment (crycode-de) [inline http-server-upload.js]: Why this variable and not just a fixed ID in the HTML part?

File: http-server-upload.js [hunk 2]
```diff
@@ -217,8 +218,8 @@ server.on('request', (req, res) => {
   <title>http-server-upload</title>
 </head>
 <body>
-<form action="upload" method="post" enctype="multipart/form-data">
-  Files: <input type="file" name="uploads" multiple="multiple"><br />
+<form action="upload" method="post" enctype="multipart/form-data" onsubmit="let totalSize=0; for (let file of document.getElementById('${formId}').files){totalSize+=file.size} if(totalSize){return (totalSize<${maxFileSize})?true:(alert(\`Cannot upload. Input files \${(totalSize/1024/1024).toFixed(2)} MB exceed ${maxFileSize/1024/1024} MB limit.\`),false) }else{alert('No file selected.');return false}">
```
Comment (crycode-de) [inline http-server-upload.js]: What about uploading files with a size of 0 bytes?
In some cases 0 bytes files may be used for signaling.

**Review by crycode-de — CHANGES_REQUESTED** [review]
Thank you for the PR!

Some remarks...

File: http-server-upload.js (hunk 1 above)
Comment (savsch) [reply http-server-upload.js]: I did that to highlight it isn't just a throwaway id, but yeah, that was too obvious.

File: http-server-upload.js (hunk 2 above)
Comment (savsch) [reply http-server-upload.js]: :facepalm: missed that

**Review by crycode-de — APPROVED** [review]

**Comment by crycode-de** [comment]
Thank you!

Just released v2.1.2

**Pull request merged**